### PR TITLE
Provide Python container path in xlang postcommit

### LINKED ISSUE #26907: [Bug]: Wheels Python SDK location does not update to the new release
### What happened?

[beam_PostCommit_Python_Xlang_Gcp_Dataflow](https://ci-beam.apache.org/job/beam_PostCommit_Python_Xlang_Gcp_Dataflow/) tests started failing when release 2.48.0 was cut with the following error:
```
RuntimeError: Pipeline construction environment and pipeline runtime environment are not compatible. 
If you use a custom container image, check that the Python interpreter minor version and the Apache 
Beam version in your image match the versions used at pipeline construction time. 
Submission environment: beam:version:sdk_base:apache/beam_python3.8_sdk:2.49.0.dev. 
Runtime environment: beam:version:sdk_base:apache/beam_python3.8_sdk:2.48.0.dev.
```

This is weird because the sdk location provided at runtime is labeled `2.49.0.dev0`:
`--sdk_location=/home/jenkins/jenkins-slave/workspace/beam_PostCommit_Python_Xlang_Gcp_Dataflow_PR/src/sdks/python/build/apache_beam-2.49.0.dev0-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`

These tests get their sdk_location at runtime via wheels:
https://github.com/apache/beam/blob/31c3390bbb4c7e6f1fc7f2c366b20d49556a61c8/sdks/python/test-suites/dataflow/common.gradle#L40-L52

After deleting that portion of code to force using tarball, the tests pass.

I've noticed that other xlang tests haven't run into this and suspect if this is because they don't use wheels to get sdk_location:
Load tests only use tarball:
https://github.com/apache/beam/blob/675d509967b7dc988aefb54181319fec237d4d06/sdks/python/apache_beam/testing/load_tests/build.gradle#L42-L45

XVR tests use the following:
https://github.com/apache/beam/blob/0963cf0bd0a791f86c1353ad488fe28f99cdfd0c/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L2597-L2603

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #26907

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by Abacn** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by Abacn** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by Abacn** [comment]
Working for direct:

before:

INFO     root:environments.py:376 Default Python SDK image for environment is apache/beam_python3.8_sdk:2.49.0.dev

now:

12:37:59 INFO     root:environments.py:376 Default Python SDK image for environment is apache/beam_python3.7_sdk:2.49.0.dev

(it should run on Py3.7 and 3.11)

**Comment by Abacn** [comment]
R: @ahmedabu98

File: .test-infra/jenkins/job_PostCommit_Python_CrossLanguage_Gcp_Direct.groovy [hunk 1]
```diff
@@ -44,7 +44,6 @@ PostcommitJobBuilder.postCommitJob('beam_PostCommit_Python_Xlang_Gcp_Direct',
           rootBuildScriptDir(commonJobProperties.checkoutDir)
           tasks(":sdks:python:test-suites:direct:gcpCrossLanguagePostCommit")
           commonJobProperties.setGradleSwitches(delegate)
-          switches("-PuseWheelDistribution")
```
Comment (ahmedabu98) [inline .test-infra/jenkins/job_PostCommit_Python_CrossLanguage_Gcp_Direct.groovy:47]: Did you mean to leave this out?

File: sdks/python/test-suites/dataflow/common.gradle [hunk 2]
```diff
@@ -458,7 +458,8 @@ project(":sdks:python:test-suites:xlang").ext.xlangTasks.each { taskMetadata ->
         "--runner=TestDataflowRunner",
         "--project=${dataflowProject}",
         "--region=${dataflowRegion}",
-        "--sdk_harness_container_image_overrides=.*java.*,gcr.io/apache-beam-testing/beam-sdk/beam_java8_sdk:latest"
+        "--sdk_harness_container_image_overrides=.*java.*,gcr.io/apache-beam-testing/beam-sdk/beam_java8_sdk:latest",
+        "--sdk_harness_container_image_overrides=.*python.*,gcr.io/apache-beam-testing/beam-sdk/beam_python${project.ext.pythonVersion}_sdk:latest"
```
Comment (ahmedabu98) [inline sdks/python/test-suites/dataflow/common.gradle:462]: --sdk_container_image for python container?

File: .test-infra/jenkins/job_PostCommit_Python_CrossLanguage_Gcp_Direct.groovy (hunk 1 above)
Comment (Abacn) [reply .test-infra/jenkins/job_PostCommit_Python_CrossLanguage_Gcp_Direct.groovy:47]: This flag was not effective at the first place. It's Direct runner, and the flag only affects dataflow runner

File: sdks/python/test-suites/dataflow/common.gradle (hunk 2 above)
Comment (Abacn) [reply sdks/python/test-suites/dataflow/common.gradle]: done (both are equivalent in this case)

**Comment by Abacn** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Review by ahmedabu98 — APPROVED** [review]
LGTM

**Comment by Abacn** [comment]
Run Spotless PreCommit

**Comment by Abacn** [comment]
Run Java_Examples_Dataflow_Java17 PreCommit

**Comment by Abacn** [comment]
Java PreCommit flink test known flakes unrelated

**Pull request merged**